### PR TITLE
update: Add a popover to prevent accidental block removal

### LINKED ISSUE #651: Provet accidental page deletion. A misclick permanently deletes page sections
following #343 and expanding one of the requests in #555 to its own bug:

As an editor and community admin, pages are _volatile_ elements to have.  

- A misclick can permanently delete an entire section.  
- Accidentally removing the wrong section can permanently delete a very full and customized section rather than a newly-created empty one.  
- Removing a section gives no warning to the clicker, so you might not notice that you've even successfully removed it.  
- If you are removing a section that extends below the bottom of the screen, and there is another section below it, you can mistakenly think that your click was not received, and click again, deleting a second or third section.
- As there is no warning given, if a section is removed unintentionally, unless the editor remembers to check the full resulting page (perhaps they went in to make a small textual edit, clicked along the side of the screen at some point and accidentally removed a long section below) they may not discover the deletion.  
- As there is no edit history, if your site hasn't been archived by the Wayback Machine, you also have no record of what disappeared and what you now need to recreate, and must do it from memory.  

As a result, while I would trust community admins to create, modify, or delete pubs, at present I don't want anyone to be able to edit Pages on communities I administer (often including myself).

Quick fix options: 
~ require confirmation before deleting a section.   
~ add a visual notification, a popup or colored bar at the top of the screen, indicating that a section has been deleted (and offering a time-limited undo)

### PR DESCRIPTION
**Pull request by qweliant**
Resolves #651 

This update addresses accidental block deletion. When removing a block, it can be easy to accidentally delete another one because of how fast the block removal happens. It can also be easy to accidentally delete a block when you are not trying to. 

old block removal
Screen Shot 2021-06-24 at 6 09 34 PM

updated block removal
Screen Shot 2021-06-24 at 5 48 50 PM

Screen Shot 2021-06-24 at 5 55 34 PM

testing plan:
test for different screen sizes 
make sure the blocks are actually removed

### CODE REVIEW

**Comment by qweliant** [comment]
would a red remove button be better or outside? similar to the page deletion button?

**Review by idreyn — CHANGES_REQUESTED** [review]
Substantially this looks great — the only thing that needs to be changed is the `aria-label` on the Popover. Nice work!

File: client/components/LayoutEditor/LayoutEditor.tsx [hunk 1]
```diff
@@ -97,7 +99,28 @@ const LayoutEditor = (props: Props) => {
...
-					{!cannotRemove && <Button text="Remove" onClick={() => removeBlock(index)} />}
+					{!cannotRemove && (
+						<>
+							<Popover
+								aria-label="Limit number of Pubs"
+								content={
+									<>
+										<Card interactive={true} elevation={Elevation.TWO}>
+											<p>Would you like to remove this block?</p>
+											<Button
```
[the remark this answers is not in the document]
Comment (qweliant) [reply client/components/LayoutEditor/LayoutEditor.tsx:110]: mmm, i can taste the fear now

**Review by idreyn — APPROVED** [review]
LGTM!

**Pull request merged**